README.md

```diff
@@ -1 +1,243 @@
+- `dev:run` - Builds the application, starts the server, launches the Finsemble application, and watches for file changes (calls "prod:run" task then the `watchFiles` function).
+- `default` - Specifies the default task to run if no task is passed in (calls the "dev:run" task).
+
+If you wish to add additional task or function calls to the existing tasks, they can be redefined in the `post` method call.
+
+Below is code that can be used as a template for _gulpfile-extensions.js_:
+
+```js
+module.exports = taskMethods => {
+    // Place your code here
+}
+```
+
+If the _gulpfile-extensions.js_ file exists, it's contents are called as a method passing an object with the default task functions in. These methods can be overwritten or extended as needed. 
+
+Example _gulpfile-extensions.js_:
+```js
+module.exports = taskMethods => {
+    "use strict";
+
+    const gulp = require("gulp");
+
+    // Extension of pre method;
+    let pre = taskMethods.pre 
+    taskMethods.pre = done => {
+        console.log("PRE: pre");
+        const result = pre(() => {
+            console.log("POST: pre");
+            done();
+        });
+    };
+
+    // Extension of copyStaticFiles
+    let copyStaticFiles = taskMethods.copyStaticFiles 
+    taskMethods.copyStaticFiles = () => {
+        console.log("PRE: copyStaticFiles");
+        const result = copyStaticFiles();
+        console.log("POST: copyStaticFiles");
+        
+        return result;
+    };
+
+    const post = taskMethods.post;
+    taskMethods.post = done => {
+        console.log("Adding tasks");
+
+        gulp.task("newTask", gulp.series("build"), () => { console.log("Do some other stuff here");});
+
+        /**
+         * Redefinition of a task defined in the gulpfile.
+         */
+        gulp.task(
+            "build",
+            gulp.series(
+                done => { console.log("pre-build task"); done(); },
+                "clean",
+                taskMethods.copyStaticFiles,
+                taskMethods.buildWebpack,
+                taskMethods.buildSass),
+                done => { console.log("post-build task"); done(); }
+            );
+
+        done();
+    }
+};
+```
+
+## Extending server functionality
+
+Sometimes during development, it is necessary to create new API for your application. The Finsemble seed project already includes a server to host files locally for development, so it makes sense to use this server to design the new API. To allow for the extension of the server provided in the Finsemble seed project without breaking the upgrade path, the server attempts to import the _server-extensions.js_ file from the _server_ directory of the project. If _server-extensions.js_ exists, it's methods are called by the server to provide additional functionality. The _server-extensions.js_ file should return an object that contains the following functions:
```

Comment:
I don't think this sentence is necessary. APIs don't need to belong on servers.

Reply:
I'm thinking a web API (like REST) that the application uses to fetch information from the server. If you don't think this makes sense, I'll remove it.